echo_server.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+import socket
+
+def response_ok():
+    return u"HTTP/1.1 200 OK\r\n \
+    Content-Type: text/plain\r\n\r\n".encode('utf-8')
+
+def response_error(error_code, reason):
+    return u"HTTP/1.1 {} {}\r\n \
+    Content-Type: text/plain\r\n\r\n".format(error_code, reason).encode('utf-8')
+
+def parse_request(request):
+    first_line = request.splitlines()[0]
+    first_line = first_line.split(' ')
+    method = first_line[0]
+    uri = first_line[1]
+    protocol = first_line[2]
+
+    if method != 'GET':
+        return response_error('405', 'Method Not Allowed')
+    elif protocol != 'HTTP/1.1':
+        return response_error('505', 'HTTP Version Not Supported')
+    else:
+        return uri
```

Comment:
As written, this method will return a URI (which is a string) if it works, and a response (which is also a string, but not the same kind of string) if it fails.  That's going to make it hard to use this since it is difficult to tell from outside the function if it worked or failed.